dirsrvtests/tests/suites/schema/schema_reload_test.py

```diff
@@ -25,6 +30,145 @@
 INVALID_SCHEMA = 'givenName $ cn $ MoZiLLaATTRiBuTe'
 
 
+def test_schema_reload_with_searches(topo):
+    """Test that during the schema reload task there is a small window where the new schema is not loaded
+       into the asi hashtables - this results in searches not returning entries.
+
+    :id: 375f1fdc-a9ef-45de-984d-0b79a40ff219
+    :setup: Standalone instance
+    :steps:
+        1. Create a test user
+        2. Run a schema_reload task while searching for our user
+        3. While we wait for the task to complete search for our user
+        4. Check the user is still being returned and if task is complete
+    :expectedresults:
+        1. Operation should be successful
+        2. Operation should be successful
+        3. Operation should be successful
+        4. Operation should be successful
+    """
+
+    log.info('Test the searches still work as expected during schema reload tasks')
+
+    # Add a user
+    users = UserAccounts(topo.standalone, DEFAULT_SUFFIX)
+    user = users.create_test_user(uid=1)
+
+    # Run a schema_reload tasks while searching for our user.Since
+    # this is a race condition, run it several times.
+
+    topo.standalone.tasks.schemaReload(schemadir=topo.standalone.schemadir,
```

Comment:
As I mentioned before, we better replace legacy objects sooner than later.
Please, use `Schema.reload` option that I've described two comments before (you can check in your email notifications because it was removed from here as you've rebased)